src/native.js

```diff
@@ -22,16 +67,45 @@ const SUPPORTED_TYPES = {
   image: IMAGE
 };
 
+const { NATIVE_KEYS, NATIVE_ASSET_TYPES, NATIVE_IMAGE_TYPES, PREBID_NATIVE_DATA_KEYS_TO_ORTB, NATIVE_KEYS_THAT_ARE_NOT_ASSETS } = CONSTANTS;
+
+// Asset type mapping as per Native IAB spec 1.2
+// https://www.iab.com/wp-content/uploads/2017/04/OpenRTB-Native-Ads-Specification-Draft_1.2_2017-04.pdf#page=40
+const assetTypeMapping = {
+  'image': {
+    1: 'icon',
+    3: 'image'
+  },
+  'data': {
+    1: 'sponsoredBy',
+    2: 'body',
+    3: 'rating',
+    4: 'likes',
+    5: 'downloads',
+    6: 'price',
+    7: 'salePrice',
+    8: 'phone',
+    9: 'address',
+    10: 'body2',
+    11: 'displayUrl',
+    12: 'cta',
+  }
+}
+
+export const nativeMapper = new Map();
```

Comment:
This should be a WeakMap (using requests as key).

Reply:
You could, then use `adapterManager.index`, as to not introduce another memory leak.

Reply:
As we settled in chat: we decided to leave the nativeMapper as is. it would take a lot of work and we'd have to change some core files in order to remove it. The goal was to not introduce memory leaks, and we temporarly "solved" by deleting native ads that are used or expired (see `removeExpiredBidsFromNativeMapper()`).